src/DotVVM.Samples.Common/TextFileDiagnosticsInformationSender.cs

```diff
@@ -0,0 +1,68 @@
+﻿using DotVVM.Framework.Diagnostics;
+using DotVVM.Framework.Configuration;
+using System.Threading.Tasks;
+using DotVVM.Framework.Diagnostics.Models;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using DotVVM.Framework.Utils;
+
+namespace DotVVM.Samples.Common
+{
+    public class TextFileDiagnosticsInformationSender : IDiagnosticsInformationSender
+    {
+        private readonly DotvvmConfiguration config;
+        private readonly string logFilePath;
+
+        public TextFileDiagnosticsInformationSender(DotvvmConfiguration config)
+        {
+            this.config = config;
+
+            logFilePath = GetFilePath();
+        }
+
+        public Task SendInformationAsync(DiagnosticsInformation information)
+        {
+            var messages = FormatUnwrittenMessages(information);
+
+            if (!File.Exists(logFilePath))
+            {
+                messages = $@"{"Event Name",-70}{"Duration [ms]",15}{"Total [ms]",15}
+{new string('-', 70 + 15 + 15)}
+{messages}";
+            }
+            File.AppendAllText(logFilePath, messages);
```

Comment:
This may run in parallel, we should have a lock around this line (I know it's just samples, but still)

Reply:
You're right - fixed.